src/btree/bt_split.c

```diff
@@ -505,9 +516,21 @@ __split_root(WT_SESSION_IMPL *session, WT_PAGE *root)
 	uint64_t split_gen;
 	uint32_t children, chunk, i, j, remain;
 	uint32_t slots;
-	bool complete;
 	void *p;
 
+	/*
+	 * A note on error handling: this function first allocates/initializes
+	 * new structures; failures during that period are handled by discarding
+	 * the memory and returning an error code, our caller knows the split
+	 * didn't happen and proceeds accordingly. Second, this function updates
+	 * the tree, and a failure in that period is catastrophic, any partial
+	 * update to the tree requires a panic, we can't recover. Third, once
+	 * the split is complete and the tree has been fully updated, we have to
+	 * ignore most errors because the split is complete and correct, callers
+	 * have to proceed accordingly.
+	 */
+	enum { WT_SPLIT_NONE, WT_SPLIT_PANIC, WT_SPLIT_IGNORE } complete;
```

Comment:
Minor nit: I think these would be clearer as `WT_SPLIT_ERR_RETURN, WT_SPLIT_ERR_PANIC, WT_SPLIT_ERR_IGNORE`